extension/source/Controllers/Keyring/KeyringController.ts

```diff
@@ -33,6 +34,33 @@ export default class KeyringController
     return this.state.wallets.map((x) => x.address);
   }
 
+  setHDPhrase(phrase: string) {
+    this.update({
+      HDPhrase: phrase,
+    });
+  }
+
+  isOnboardingComplete = (): boolean => {
+    return this.state.HDPhrase !== '';
+  };
+
+  async createHDAccount(): Promise<string> {
+    const mnemonic = this.state.HDPhrase;
+
+    if (mnemonic === '') {
+      const { phrase } = ethers.Wallet.createRandom().mnemonic;
+      this.setHDPhrase(phrase);
```

Comment:
It looks like you intended `mnemonic` to be `phrase` after this point but forgot to set it.

Reply:
Ah you are right, I updated the state but did not refetch the new state. Will fix.

Reply:
Fixed.